src/Sarif/ExtensionMethods.cs

```diff
@@ -585,5 +586,15 @@ internal static void ConsumeElementOfDepth(this XmlReader xmlReader, int endElem
                 xmlReader.Read();
             }
         }
+
+        public static void Add<T>(this ConcurrentDictionary<string, T> concurrentDictionary, string key, T value)
```

Comment:
>ConcurrentDictionary [](http://example.com/codeflow?start=39&length=20)

Why do we need these? Can't we just require our code to cast to IDictionary when appropriate?